### DISCUSSION TITLE
What can we do to improve the docs on UWP migration?

### DISCUSSION THREAD

**Discussion by bgursahani** [microsoft/WindowsAppSDK#1705] [Feedback wanted]
Hey folks,

The announcement about UWP and the Windows App SDK went live about two weeks ago. Here it is: https://github.com/microsoft/WindowsAppSDK/discussions/1615

In addition, the UWP migration documentation is now live, and you can find it here: [Migrate to the Windows App SDK](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/migrate-to-windows-app-sdk-ovw) and we would like to receive feedback on the docs. 

We’re looking forward to receiving your feedback and continuing to improve the docs.

1. Please take your time and try out the migration journey for yourself.
2. Please let us know your feelings about the docs.
3. Please provide feedback by replying directly to this discussion. Please include:
    a.	What you liked about the docs.
    b.	What you didn’t like.
    c.	And what you’d suggest improving.

The docs are structured to make migration easier for both new and experienced developers. Here’s the table of contents:

- Migrate to the Windows App SDK

    - Index of migration topics

    - Overall migration strategy

    - Mapping UWP features to Windows App SDK

    - What’s supported

    - Mapping UWP APIs to Windows App SDK

    - Feature area guides

        - Overview of feature area guides

        - Windows UI Library (WinUI) migration

        - Windowing functionality migration

        - Application lifecycle functionality migration

        - Notifications migration

        - DirectWrite to DWriteCore migration

        - MRT to MRTCore migration

        - Threading functionality migration

    - Case study 1—PhotoLab (C#)

    - Case Study 2—PhotoEditor(C++/WinRT)

    - Additional migration guidance

1.	[Overall migration strategy ](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/overall-migration-strategy)puts forward a value proposition and framework to start your migration journey. Whereas [What’s supported ](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/what-is-supported)advises you whether you should begin your migration journey now, or wait for a future release of the Windows App SDK.
2.	The [Feature area guides](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/guides/feature-area-guides-ovw) section takes a deeper dive into how you can migrate specific feature areas of your application.
3.	The feature and API mapping tables bring clarity and provide easy access to all of the available mapping information.
4.	The two case studies (C# and C++/WinRT) are step-by-step walkthroughs that give you an immediate experience of the migration journey.

Thank you

**Comment by myokeeh** [comment]
I asked a specific question here https://github.com/microsoft/WindowsTemplateStudio/issues/4387 since I use WindowsTemplateStudio. For someone who rely on such tools to get from zero to working app, everything about WindowsAppSDK still feels very uncertain. Lack of answers/activity there doesn't bode well. Maybe I'm wrong. If a tool like WindowsTemplateStudio can't be brought along quickly along side WindowsAppSDK, it doesn't instill confidence.

**Reply by bgursahani** [reply]
Thanks for the feedback. We're currently still in our early stages of Windows App SDK and will only be releasing version 1.0 next week. The Windows Template Studio indeed doesn't support WinUI 3 apps right now. We're considering supporting the Windows Template Studio but it's not on our immediate plans yet, you can vote for that here and your vote can help change the priority: [Windows Template Studio - Windows App SDK | Product Roadmap (productboard.com)](https://portal.productboard.com/winappsdk/1-windows-app-sdk/c/52-windows-template-studio)

**Comment by MattBDev** [comment]
I think it should be mentioned in the migration guide that one reason not to migrate from UWP to the Windows App SDK is that it isn't supported on devices like XBox, IoT, Surface Hub, and Hololens. I think it makes sense to mention it somewhere since Windows Desktop apps seem to be moving forward without other Windows devices in mind.

**Comment by pierresa** [comment]
UWP comes with loads of examples (especially [Github: Win](https://github.com/Microsoft/Windows-universal-samples)) You need to migrate them to convince us that migtation is feasible/usefull/not an other waste of time.

The set of exemaples currently provided (https://github.com/microsoft/WindowsAppSDK-Samples) is too poor/limited
